### PR TITLE
Fix Sonar issues Cli and CliAutoComplete

### PR DESCRIPTION
**Pull request by McGiverGim**
Cleaning of major part of Sonar issues in CLI and the CLIAutoComplete parts.

### CODE REVIEW

File: src/js/CliAutoComplete.js [hunk 1]
```diff
@@ -193,55 +203,57 @@ CliAutoComplete.builderParseLine = function(line) {
...
+    const highlighterAnywhere = highlighter(true);
+    const highlighterPrefix = highlighter(false);
 
-    var searcher = function(term, callback, array, minChars, matchPrefix) {
-        var res = [];
+    const searcher = function(term, callback, array, minChars, matchPrefix) {
+        const res = [];
 
-        if ((minChars !== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
+        if ((minChars!== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
```
Comment (haslinghuis) [inline src/js/CliAutoComplete.js:227]:
```suggestion
        if ((minChars !== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
```
Is this intentional?

File: src/js/CliAutoComplete.js [hunk 2]
```diff
@@ -193,55 +203,57 @@ CliAutoComplete.builderParseLine = function(line) {
...
             term = term.toLowerCase();
-            for (var i = 0; i < array.length; i++) {
-                var v = array[i].toLowerCase();
-                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term) !== -1) {
+            for (let i = 0; i < array.length; i++) {
+                const v = array[i].toLowerCase();
+                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term)!== -1) {
```
Comment (haslinghuis) [inline src/js/CliAutoComplete.js:231]:
```suggestion
                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term) !== -1) {
```
Same.

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: src/js/CliAutoComplete.js (hunk 1 above)
Comment (McGiverGim) [reply src/js/CliAutoComplete.js]: Fixed!

File: src/js/CliAutoComplete.js (hunk 2 above)
Comment (McGiverGim) [reply src/js/CliAutoComplete.js]: Fixed!

**Comment by McGiverGim** [comment]
Fixed as per review (thanks @haslinghuis !)

**Comment by McGiverGim** [comment]
Rebased against master, to get the real number of code smells...

**Review by haslinghuis — APPROVED** [review]
Thanks. LGTM!

**Review by mikeller — DISMISSED** [review]

**Comment by mikeller** [comment]
Weirdly Travis tests are failing for this change.

**Comment by McGiverGim** [comment]
I have moved the obtention of some GUI elements to a method that the tests do not call and for this reason they are failing. I will look at this later, I have not experience with this Karma thing...

**Comment by McGiverGim** [comment]
Fixed, but I'm not an expert in this Karma thing so maybe I have made some mistake:
- I have modified the tests to have all the elements that I reordered, so the tests can be executed.
- I have added the `spec` reporter, because I was not able to see the result of the tests. Maybe this is wrong and I need to remove it? I have not removed the old `tfs` because maybe is used at some place unknown to me...
- Fixed some Sonar issues in this tests files too...
- Rebased against master...

File: package.json [hunk 3]
```diff
@@ -55,8 +55,7 @@
     "inflection": "^1.12.0",
     "jbox": "^1.0.8",
     "jquery": "^3.5.1",
-    "jquery-textcomplete": "^1.8.5",
-    "jquery-touchswipe": "^1.6.19",
+    "jquery-textcomplete": "^1.8.5",    "jquery-touchswipe": "^1.6.19",
```
Comment (mikeller) [inline package.json:58]: Not sure what happened here.

**Review by mikeller — APPROVED** [review]
Apart from the weirdness in `package.json` this looks good to me.

File: package.json (hunk 3 above)
Comment (McGiverGim) [reply package.json]: Me neither... it's clear that I'm not at my best moment 🤸‍♂️

**Comment by McGiverGim** [comment]
Now we have this output in the tests:

image

I don't know if is ok for our Jenkins output or we do prefer to remove it.

**Review by mikeller — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]
🤗

**Pull request merged**